src/Workspaces/Core/Portable/Execution/Serialization/Checksum.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Execution
+{
+    /// <summary>
+    /// Checksum of data
+    /// </summary>
+    internal sealed partial class Checksum : IObjectWritable, IEquatable<Checksum>
```

Comment:
Can this just be a struct?  Why do we need an allocation just to wrap a byte[] ?